bql/semantic/expression_test.go

```diff
@@ -636,84 +619,102 @@ func TestNewEvaluator(t *testing.T) {
 			r: table.Row{
 				"?o": &table.Cell{L: mustBuildLiteral(`"73"^^type:int64`)},
 			},
-			err:  false,
 			want: false,
 		},
 		{
-			id: `?s ID ?id > "37"^^type:int64`,
+			id: `?foo = "10"^^type:text`,
 			in: []ConsumedElement{
 				NewConsumedToken(&lexer.Token{
 					Type: lexer.ItemBinding,
-					Text: "?id",
+					Text: "?foo",
 				}),
 				NewConsumedToken(&lexer.Token{
-					Type: lexer.ItemGT,
+					Type: lexer.ItemEQ,
 				}),
 				NewConsumedToken(&lexer.Token{
 					Type: lexer.ItemLiteral,
-					Text: `"37"^^type:int64`,
+					Text: `"10"^^type:text`,
 				}),
 			},
 			r: table.Row{
-				"?id": &table.Cell{S: table.CellString("peter")},
+				"?foo": &table.Cell{L: mustBuildLiteral(`"10"^^type:int64`)},
 			},
-			err:  true,
 			want: false,
 		},
+	}
+	for _, entry := range testTable {
+		eval, err := NewEvaluator(entry.in)
+		if err != nil {
+			t.Fatalf("test %q should have never failed when creating a new evaluator from %v, got error: %v", entry.id, entry.in, err)
+		}
+
+		got, err := eval.Evaluate(entry.r)
+		if err != nil {
+			t.Errorf("%s: eval.Evaluate(%v) = _, %v; want nil error", entry.id, entry.r, err)
+		}
+		if want := entry.want; got != want {
+			t.Errorf("%s: eval.Evaluate(%v) = %v, _; want %v, _", entry.id, entry.r, got, want)
+		}
```

Comment:
This function is called `TestNewEvaluator`, which indicates that it is testing the `NewEvaluator` func. However, it seems it is actually testing `eval.Evaluate` and thus should be named something like `TestEvaluatorEvaluate` or something in those lines. 
Is that the case or there's something I'm not seeing here?

Reply:
I agree with you and I have just renamed the function and its correspondent error checking one. Thank you once again Tati. 😃